src/pages/PageAudioCall.tsx

```diff
@@ -1,15 +1,44 @@
 import React, { useState } from "react";
-import { useHistory, useParams } from "react-router-dom";
+import {
+  useHistory,
+  useLocation,
+  useParams,
+  RouteComponentProps,
+} from "react-router-dom";
+
 import { Helmet } from "react-helmet";
 
 import { useAudioSdk, useClient, useMicrophoneTrack } from "../hooks/audioSdk";
 import type { IMicrophoneAudioTrack } from "agora-rtc-sdk-ng";
+import {
+  EVENT_STREAM_PUBLISHED,
+  EVENT_STREAM_UNPUBLISHED,
+} from "../constants/events";
+import {
+  uniqueNamesGenerator,
+  Config,
```

Comment:
I dislike how they name it `Config`, like its too general and can be easily misunderstood. Try this instead
```suggestion
  Config as UniqueNamesConfig,
```